### PR TITLE
removing deployment service class

### PR DESCRIPTION
**Pull request by srijitamukherjee**
**What this PR does / why we need it**:

Removing ApicastV1 Deployment service class

**Which issue(s) this PR fixes** 

 part of [THREESCALE-7134](https://issues.redhat.com/browse/THREESCALE-7134)

### CODE REVIEW

File: test/unit/proxy_test.rb [hunk 1]
```diff
@@ -478,17 +478,6 @@ def test_apicast_configuration_driven
     end
   end
 
-  test 'save_and_deploy' do
```
Comment (hallelujah) [inline test/unit/proxy_test.rb:481]: Do we have another test for apicast v2 or service mesh?

Comment (srijitamukherjee) [reply test/unit/proxy_test.rb]: no there is not.

File: test/unit/proxy_test.rb [hunk 2]
```diff
@@ -483,8 +483,13 @@ def test_apicast_configuration_driven
                               api_backend: 'http://example.com',
                               api_test_path: '/path',
                               apicast_configuration_driven: false)
-    ::ApicastV1DeploymentService.any_instance.expects(:deploy).with(proxy).returns(true).never
+    ::ApicastV1DeploymentService.any_instance.expects(:call).returns(true)
```
Comment (hallelujah) [inline test/unit/proxy_test.rb:486]: This will fail if you removed it no?
Replace it with the equivalent of v2 if it does not exist already

Comment (HemaHG) [reply test/unit/proxy_test.rb]: since you are using Apicast, change `apicast_configuration_driven: false` to `true`

**Review by hallelujah — DISMISSED** [review]
🚀

**Review by hallelujah — DISMISSED** [review]

**Review by hallelujah — APPROVED** [review]

**Review by jsmadis — APPROVED** [review]

**Pull request merged**